UnitySetup/UnitySetup.psm1

```diff
@@ -2132,3 +2130,537 @@ function Get-UnityLicense {
         Write-Warning "Alias $($_.Name) already configured by $($alias.Source)"
     }
 }
+
+function Import-UPMConfig {
+    [CmdletBinding()]
+    param(
+        [Parameter()]
+        [String]$ProjectManifestPath,
+        [int]$SearchDepth = 3
+    )
+
+    $ProjectManifestPaths = @()
+
+    if ((Get-Item $ProjectManifestPath) -is [System.IO.DirectoryInfo]) {
+        Write-Host "Path provided is not a manifest.json file ($ProjectManifestPath), will attempt search"
+        $FoundPaths = @(Get-Childitem -Path $ProjectManifestPath -Include manifest.json -File -Recurse -Depth $SearchDepth -ErrorAction SilentlyContinue)
+        foreach ($file in $FoundPaths) {
+            $ProjectManifestPaths += $file.FullName
+            if ($Verbose) { Write-Host "Found ($file.FullName)" }
+        }
+    }
+    else {
+        $ProjectManifestPaths += $ProjectManifestPath
+    }
+
+    if (([string]::IsNullOrEmpty($ProjectManifestPath)) -or (-not $(Test-Path $ProjectManifestPath))) {
+        Write-Error "Unable to find manifest.json file, please provide a path pointing directly to a Unity project's manifest.json or provide a path to a Unity project"
+        return @()
+    }
+
+    $scopedRegistrySet = [System.Collections.Generic.HashSet[string]]::new()
+    foreach ($ManifestPath in $ProjectManifestPaths) {
+        $manifest = Get-Content -Path $ManifestPath | ConvertFrom-Json
+
+        foreach ($scopedRegistry in $manifest.scopedRegistries) {
+            $url = $scopedRegistry.url -replace '/$', ''  
+            if ($url -like 'https://pkgs.dev.azure.com/*') {
+                $scopedRegistrySet.Add($url) | Out-Null
+            }
+        }
+    }
+
+    return [System.Collections.Generic.HashSet[string]]::new($scopedRegistrySet)
```

Comment:
Would prefer I think if we returned the whole manifest from this cmdlet - users who only care about scoped registries could then investigate that.

Reply:
Sounds good - though I'll need to make it return an array of manifests as opposed to just one, since the function scans and aggregates scoped registries.  We can push the hashset deduping of registries further down the process to Sync cmdlet